File: golang/num_guess/num_guess.go
package main

import (
	"fmt";
	"strconv";
	"math/rand" )

func main() {
    fmt.Println("Guess a number!")

	var guess string
	max_tries :=3
	answer := rand.Intn(11)  

	for {
		if max_tries == 0{
			fmt.Println("You have reached max tries of 3 times")
			return
		}

		fmt.Print("Please enter an integer. (below 10): ")  
		fmt.Scanln(&guess)
		
	
		guessInt, err := strconv.Atoi(guess)  
		if err != nil {  
			fmt.Println("Please enter a valid integer")
			max_tries -=1 
		} else if guessInt > 11 {
			fmt.Println("Please enter a number below 10")
			max_tries -=1
		} else {
			if _compare_val(&guessInt,&answer) {
				return;
			}
			
		}
	}

}

func _compare_val(guess *int,answer *int) bool {


	// fmt.Println("Correct number is:", answer)  
	fmt.Println("You guessed: ",*guess)
	// fmt.Printf("You guessed: %s,Correct answer is: %s. ", guess, answer)  
	
	if *guess == *answer {
		fmt.Println("Congrats, You are correct!")
		return true
	} else {
		// fmt.Println("Sorry you are wrong.")
		if *guess > *answer {
			fmt.Println("Guess too big!")
		} else {
			fmt.Println("Guess too small!")
		}
		return false
	}
	return false
}

File: julia/num_guess/src/num_guess.jl
using Random  
  
function guessNum()
    println("Guess the number!")  

    max_tries =3 
    answer = rand(1:10) 
    while true

        if max_tries == 0 
            println("Sorry, you have reached max tries of 3 times.")
            return
        end

        print("Please enter a number (below 10): ")    
        guess = readline() 
        guessInt = tryparse(Int, guess) 

        if isnothing(guessInt)  
            println("Please enter a valid integer")
            max_tries -=1 
        else   

            if guessInt > 10  
                println("Please enter a number below 10!")
                max_tries -=1
            else  
                if _compare_vals(guessInt,answer)  
                    return    
                end
            end 

        end  

    end

end  


function _compare_vals(guess,answer)  
   
    println("You guessed: ", guess)  
    # println("Correct number is: ", correct) 
    # println("You guessed: $(guess)!, correct is: $correct.")  
    
    if answer == guess 
        println("Congrats, you guessed corectly")
        return true
    else
        # println("Sorry, you are wrong!") 
        if guess > answer 
            println("Guess too big!")
        else
            println("Guess too small")
        end
        return false
    end
end 
  



guessNum()